### PR TITLE
Add flake.nix

### PR DESCRIPTION
**Pull request by tadfisher**
Add `flake.nix` with `defaultPackage` and `devShell` outputs.

This should still work with normal Nix using [flake-compat](https://github.com/edolstra/flake-compat).

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          pkgs = pkgsBySystem.${system};
+        in
+          pkgs.mkShell {
+            name = "moat-shell";
+            inputsFrom = [ self.defaultPackage.${system} ];
+            buildInputs = with pkgs.haskellPackages; [
+              cabal-install
+              ghc
+              ghcid
+              haskell-language-server
```
Comment (chiroptical) [inline flake.nix:62]: Do you think it makes sense to leave this out and rely on the system version for now?

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
+{
+  description = "Generate swift and kotlin types from haskell types";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
```
Comment (chiroptical) [inline flake.nix:5]: This is pinned by `flake.lock` right?

File: flake.nix (hunk 1 above)
Comment (tadfisher) [reply flake.nix:62]: Sure, it's a pretty heavy package so I can just install it in my profile.

File: flake.nix (hunk 2 above)
Comment (tadfisher) [reply flake.nix:5]: Correct. We can add a commit hash here to pin it to something specific.

**Review by chiroptical — APPROVED** [review]

**Pull request merged**